go/lang/security/audit/database/string-formatted-query.yaml

```diff
@@ -11,6 +11,13 @@ rules:
     cwe: "CWE-89: Improper Neutralization of Special Elements used in an SQL Command ('SQL Injection')"
     source-rule-url: https://github.com/securego/gosec
   patterns:
+  - metavariable-regex:
+      metavariable: $SINK
+      regex: (Exec|ExecContext|Query|QueryContext|QueryRow|QueryRowContext)
+  - pattern-not-inside: |
+      $VAR = "..." + "..."
+      ...
+      $OBJ.$SINK(..., $VAR, ...)
   - pattern-either:
     - pattern: $OBJ.Exec("..." + $X)
```

Comment:
Can we use `$OBJ.$SINK` for the rest of these `pattern`s too, and leverage the `metavariable-regex`? That would cut this rule down by about 10x.

Reply:
We could, but a pattern with all metavariables involved tends to reduce performance. I was able to improve some JavaScript rules that performed this way by expanding the metavariable regex into distinct pattern clauses, which I suppose helps semgrep-core more easily identify which patterns to match.